pykokkos/core/parsers/parser.py

```diff
@@ -123,19 +123,186 @@ def get_entities(self, style: PyKokkosStyles) -> Dict[str, PyKokkosEntity]:
+                        value = ast.Name(id=self.pk_import, ctx=ast.Load()),
+                        attr = view_type,
+                        ctx = ast.Load()
+                    ),
+                    slice = ast.Attribute(
+                        value = ast.Name(id=self.pk_import, ctx=ast.Load()),
+                        attr = dtype,
+                        ctx = ast.Load()
+                    ),
+                    ctx = ast.Load()
+                )
+
+            elif "Acc:" in update_type:
+                dtype = update_type.split(":")[1]
+
+                arg_obj.annotation = ast.Subscript(
+                        value = ast.Attribute(
+                            value = ast.Name(id=self.pk_import, ctx=ast.Load()),
+                            attr = "Acc",
+                            ctx = ast.Load()
+                    ),
+                    slice = ast.Attribute(
+                        value = ast.Name(id=self.pk_import, ctx=ast.Load()),
+                        attr = dtype,
+                        ctx=ast.Load(),
+                    ),
+                    ctx = ast.Load()
+                )
+
+            elif "TeamMember" in update_type: #"pk.TeamMember" is hard-set in get_annotations
+                arg_obj.annotation = ast.Attribute(
+                    value = ast.Name(id=self.pk_import, ctx=ast.Load()),
+                    attr = "TeamMember",
+                    ctx = ast.Load()
+                )
+            else:
+                raise ValueError(f"Type inference for {update_type} is not supported")
+
+        assert entity_tree is not None
+        return entity_tree
+
+
+    def fix_view_layout(self, node : ast.AST, layout_change: Dict[str, str]) -> List[ast.Call]:
+        '''
+        Add (to the node AST) the missing layout decorators for pykokkos views
+
+        node: ast object for the entity
+        layout_change: Dict that maps [view -> layout]
+        '''
+
+        assert len(node.decorator_list), "Decorator cannot be missing for pykokkos workunit"
+        # Decorator list will have ast.Call object as first element if user has provided layout decorators
+        is_layout_given: bool = isinstance(node.decorator_list[0], ast.Call)
+
+        if is_layout_given:
+            # Filter out layouts already given by user
+            layout_change = self.filter_layout_change(node, layout_change)
+
+        if len(layout_change):
+            call_obj = None
+
+            if is_layout_given: # preserve the existing call object
+                call_obj = node.decorator_list[0]
+            else: 
+                call_obj= ast.Call()
+                call_obj.func = ast.Attribute(value=ast.Name(id=self.pk_import, ctx=ast.Load()), attr='workunit', ctx=ast.Load())
+                call_obj.args = []
+                call_obj.keywords = []
+
+            for view, layout in layout_change.items():
+                call_obj.keywords.append(
+                    ast.keyword(
```

Comment:
can this be extracted into a function/method

Reply:
Done